src/libraries/System.Runtime/tests/System/SingleTests.cs

```diff
@@ -727,39 +730,32 @@ public static IEnumerable<object[]> ToString_TestData_NotNetFramework()
             yield return new object[] { 32.5f, "N100", invariantFormat, "32.5000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000" };
         }
 
-        [Fact]
-        public static void Test_ToString_NotNetFramework()
+        [Theory]
+        [MemberData(nameof(ToString_TestData_NotNetFramework))]
+        public static void Test_ToString_NotNetFramework(float f, string format, IFormatProvider provider, string expected)
         {
             using (new ThreadCultureChange(CultureInfo.InvariantCulture))
             {
-                foreach (object[] testdata in ToString_TestData_NotNetFramework())
+                bool isDefaultProvider = provider == null;
+                if (string.IsNullOrEmpty(format) || format.ToUpperInvariant() == "G")
                 {
-                    ToStringTest((float)testdata[0], (string)testdata[1], (IFormatProvider)testdata[2], (string)testdata[3]);
+                    if (isDefaultProvider)
+                    {
+                        Assert.Equal(expected, f.ToString());
+                        Assert.Equal(expected, f.ToString((IFormatProvider)null));
+                    }
+                    Assert.Equal(expected, f.ToString(provider));
                 }
-            }
-        }
-
-        private static void ToStringTest(float f, string format, IFormatProvider provider, string expected)
-        {
-            bool isDefaultProvider = provider == null;
-            if (string.IsNullOrEmpty(format) || format.ToUpperInvariant() == "G")
-            {
                 if (isDefaultProvider)
                 {
-                    Assert.Equal(expected, f.ToString());
-                    Assert.Equal(expected, f.ToString((IFormatProvider)null));
+                    Assert.Equal(expected, f.ToString(format.ToUpperInvariant()), true);
```

Comment:
Consider naming opaque parameters at the call sites for readability, eg.,
```suggestion
                    Assert.Equal(expected, f.ToString(format.ToUpperInvariant()), ignoreCase: true);
```

Reply:
Fixed by [31c3ffac11476a2ca7c23be756e2cf9e18059391](https://github.com/dotnet/runtime/pull/85565/commits/31c3ffac11476a2ca7c23be756e2cf9e18059391) commit.